### PR TITLE
Add missing ckpt in config docs

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

As discussed on Slack, I worked on the `Config` files to add missing information about checkpoints, or correct them.

- I tried to check the mentioned checkpoints are actually on the Hub
- also tried to make sure the checkpoints are for the target architecture
- I didn't verify the statement `Instantiating a configuration with the defaults will yield a similar configuration to that of the Speech2Text2 [mentioned checkpoint]`
   - in particular, the hyperparameters like `hidden_dim`, `num_layers` might be different
   - it says `similar`, so I think it is fine (..?)

@patrickvonplaten Could you take a look on the speech models?
@NielsRogge Could you take a look on the vision models?

### CODE REVIEW

File: src/transformers/models/speech_to_text_2/configuration_speech_to_text_2.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ class Speech2Text2Config(PretrainedConfig):
     This is the configuration class to store the configuration of a [`Speech2Text2ForCausalLM`]. It is used to
     instantiate an Speech2Text2 model according to the specified arguments, defining the model architecture.
     Instantiating a configuration with the defaults will yield a similar configuration to that of the Speech2Text2
-    [facebook/s2t-small-librispeech-asr](https://huggingface.co/facebook/s2t-small-librispeech-asr) architecture.
+    [facebook/s2t-wav2vec2-large-en-de](https://huggingface.co/facebook/s2t-wav2vec2-large-en-de) architecture.
```
Comment (ydshieh) [inline src/transformers/models/speech_to_text_2/configuration_speech_to_text_2.py:34]: `facebook/s2t-small-librispeech-asr` has `model_type": "speech_to_text`, but here is for the model `Speech2Text2` (i.e. `speech_to_text_2`)

File: src/transformers/models/dpr/configuration_dpr.py [hunk 2]
```diff
@@ -35,7 +35,14 @@ class DPRConfig(PretrainedConfig):
...
     This is the configuration class to store the configuration of a [`DPRContextEncoder`], [`DPRQuestionEncoder`], or a
-    [`DPRReader`]. It is used to instantiate the components of the DPR model.
+    [`DPRReader`]. It is used to instantiate the components of the DPR model according to the specified arguments,
+    defining the model component architectures. Instantiating a configuration with the defaults will yield a similar
+    configuration to that of the DPRContextEncoder
+    [facebook/dpr-ctx_encoder-single-nq-base](https://huggingface.co/facebook/dpr-ctx_encoder-single-nq-base),
+    DPRQuestionEncoder
+    [facebook/dpr-question_encoder-single-nq-base](https://huggingface.co/facebook/dpr-question_encoder-single-nq-base),
+    or DPRReader [facebook/dpr-reader-single-nq-base](https://huggingface.co/facebook/dpr-reader-single-nq-base)
+    architecture.
```
Comment (sgugger) [inline src/transformers/models/dpr/configuration_dpr.py:45]: Maybe one checkpoint is enough here?

File: src/transformers/models/retribert/configuration_retribert.py [hunk 3]
```diff
@@ -22,14 +22,16 @@
 
 # TODO: upload to AWS
 RETRIBERT_PRETRAINED_CONFIG_ARCHIVE_MAP = {
-    "retribert-base-uncased": "https://huggingface.co/distilbert-base-uncased/resolve/main/config.json",
+    "retribert-base-uncased": "https://huggingface.co/yjernite/retribert-base-uncased/resolve/main/config.json",
```
Comment (sgugger) [inline src/transformers/models/retribert/configuration_retribert.py:25]: The namespace is missing in the key:
```suggestion
    "yjernite/retribert-base-uncased": "https://huggingface.co/yjernite/retribert-base-uncased/resolve/main/config.json",
```
```suggestion
    "retribert-base-uncased": "https://huggingface.co/yjernite/retribert-base-uncased/resolve/main/config.json",
```

Comment (sgugger) [inline src/transformers/models/retribert/configuration_retribert.py:25]: The namespace is missing in the key:
```suggestion
    "yjernite/retribert-base-uncased": "https://huggingface.co/yjernite/retribert-base-uncased/resolve/main/config.json",
```
```suggestion
    "retribert-base-uncased": "https://huggingface.co/yjernite/retribert-base-uncased/resolve/main/config.json",
```

File: src/transformers/models/retribert/configuration_retribert.py [hunk 4]
```diff
@@ -22,14 +22,16 @@
...
 }
 
 
 class RetriBertConfig(PretrainedConfig):
     r"""
     This is the configuration class to store the configuration of a [`RetriBertModel`]. It is used to instantiate a
-    RetriBertModel model according to the specified arguments, defining the model architecture.
+    RetriBertModel model according to the specified arguments, defining the model architecture. Instantiating a
+    configuration with the defaults will yield a similar configuration to that of the RetriBert
+    [retribert-base-uncased](https://huggingface.co/yjernite/retribert-base-uncased) architecture.
```
Comment (sgugger) [inline src/transformers/models/retribert/configuration_retribert.py:34]:
```suggestion
    [yjernite/retribert-base-uncased](https://huggingface.co/yjernite/retribert-base-uncased) architecture.
```
Likewise

Comment (sgugger) [inline src/transformers/models/retribert/configuration_retribert.py:34]:
```suggestion
    [yjernite/retribert-base-uncased](https://huggingface.co/yjernite/retribert-base-uncased) architecture.
```
Likewise

**Review by sgugger — APPROVED** [review]
Thanks for fixing all of those! It would be awesome to have some kind of quality script to check we don't introduce new faulty checkpoints.

File: src/transformers/models/luke/configuration_luke.py [hunk 5]
```diff
@@ -29,7 +29,9 @@
 class LukeConfig(PretrainedConfig):
     r"""
     This is the configuration class to store the configuration of a [`LukeModel`]. It is used to instantiate a LUKE
-    model according to the specified arguments, defining the model architecture.
+    model according to the specified arguments, defining the model architecture. Instantiating a configuration with the
+    defaults will yield a similar configuration to that of the Luke
```
Comment (NielsRogge) [inline src/transformers/models/luke/configuration_luke.py:33]:
```suggestion
    defaults will yield a similar configuration to that of the LUKE
```

File: src/transformers/models/flaubert/configuration_flaubert.py [hunk 6]
```diff
@@ -36,6 +36,8 @@ class FlaubertConfig(XLMConfig):
     """
     This is the configuration class to store the configuration of a [`FlaubertModel`] or a [`TFFlaubertModel`]. It is
     used to instantiate a FlauBERT model according to the specified arguments, defining the model architecture.
+    Instantiating a configuration with the defaults will yield a similar configuration to that of the Flaubert
```
Comment (NielsRogge) [inline src/transformers/models/flaubert/configuration_flaubert.py:39]:
```suggestion
    Instantiating a configuration with the defaults will yield a similar configuration to that of the FlauBERT
```

File: src/transformers/models/maskformer/configuration_maskformer.py [hunk 7]
```diff
@@ -36,8 +36,8 @@ class MaskFormerConfig(PretrainedConfig):
     This is the configuration class to store the configuration of a [`MaskFormerModel`]. It is used to instantiate a
     MaskFormer model according to the specified arguments, defining the model architecture. Instantiating a
     configuration with the defaults will yield a similar configuration to that of the
-    "facebook/maskformer-swin-base-ade" architecture trained on
-    [ADE20k-150](https://huggingface.co/datasets/scene_parse_150).
+    [facebook/maskformer-swin-base-ade](https://huggingface.co/facebook/maskformer-swin-base-ade) architecture trained
+    on [ADE20k-150](https://huggingface.co/datasets/scene_parse_150).
```
Comment (NielsRogge) [inline src/transformers/models/maskformer/configuration_maskformer.py:40]:
```suggestion
    MaskFormer [facebook/maskformer-swin-base-ade](https://huggingface.co/facebook/maskformer-swin-base-ade) architecture trained on [ADE20k-150](https://huggingface.co/datasets/scene_parse_150).
```

File: src/transformers/models/mobilebert/configuration_mobilebert.py [hunk 8]
```diff
@@ -21,14 +21,16 @@
...
-    "mobilebert-uncased": "https://huggingface.co/google/mobilebert-uncased/resolve/main/config.json"
+    "google/mobilebert-uncased": "https://huggingface.co/google/mobilebert-uncased/resolve/main/config.json"
 }
 
 
 class MobileBertConfig(PretrainedConfig):
     r"""
     This is the configuration class to store the configuration of a [`MobileBertModel`] or a [`TFMobileBertModel`]. It
     is used to instantiate a MobileBERT model according to the specified arguments, defining the model architecture.
+    Instantiating a configuration with the defaults will yield a similar configuration to that of the MobileBert
```
Comment (NielsRogge) [inline src/transformers/models/mobilebert/configuration_mobilebert.py:32]:
```suggestion
    Instantiating a configuration with the defaults will yield a similar configuration to that of the MobileBERT
```

File: src/transformers/models/reformer/configuration_reformer.py [hunk 9]
```diff
@@ -30,7 +30,9 @@
 class ReformerConfig(PretrainedConfig):
     r"""
     This is the configuration class to store the configuration of a [`ReformerModel`]. It is used to instantiate a
-    Reformer model according to the specified arguments, defining the model architecture.
+    Reformer model according to the specified arguments, defining the model architecture. Instantiating a configuration
+    with the defaults will yield a similar configuration to that of the Reformer
```
Comment (NielsRogge) [inline src/transformers/models/reformer/configuration_reformer.py:34]:
```suggestion
    with the defaults will yield a similar configuration to that of the ReFormer
```

**Comment by ydshieh** [comment]
> Thanks for fixing all of those! It would be awesome to have some kind of quality script to check we don't introduce new faulty checkpoints.

Yes, I do have some (draft) check locally. I plan to add it in another PR (unless it's necessary to do so in this PR).

File: src/transformers/models/regnet/configuration_regnet.py [hunk 10]
```diff
@@ -30,7 +30,7 @@ class RegNetConfig(PretrainedConfig):
     This is the configuration class to store the configuration of a [`RegNetModel`]. It is used to instantiate a RegNet
     model according to the specified arguments, defining the model architecture. Instantiating a configuration with the
     defaults will yield a similar configuration to that of the
-    [facebook/regnet-y-40](https://huggingface.co/facebook/regnet-y-40) architecture.
+    [facebook/regnet-y-040](https://huggingface.co/facebook/regnet-y-040) architecture.
```
Comment (NielsRogge) [inline src/transformers/models/regnet/configuration_regnet.py:33]:
```suggestion
    RegNet [facebook/regnet-y-040](https://huggingface.co/facebook/regnet-y-040) architecture.
```

File: src/transformers/models/resnet/configuration_resnet.py [hunk 11]
```diff
@@ -30,7 +30,7 @@ class ResNetConfig(PretrainedConfig):
     This is the configuration class to store the configuration of a [`ResNetModel`]. It is used to instantiate an
     ResNet model according to the specified arguments, defining the model architecture. Instantiating a configuration
     with the defaults will yield a similar configuration to that of the
-    [resnet-50](https://huggingface.co/microsoft/resnet-50) architecture.
+    [microsoft/resnet-50](https://huggingface.co/microsoft/resnet-50) architecture.
```
Comment (NielsRogge) [inline src/transformers/models/resnet/configuration_resnet.py:33]:
```suggestion
    ResNet [microsoft/resnet-50](https://huggingface.co/microsoft/resnet-50) architecture.
```

File: src/transformers/models/roberta/configuration_roberta.py [hunk 12]
```diff
@@ -38,7 +38,8 @@ class RobertaConfig(BertConfig):
     r"""
     This is the configuration class to store the configuration of a [`RobertaModel`] or a [`TFRobertaModel`]. It is
     used to instantiate a RoBERTa model according to the specified arguments, defining the model architecture.
-
+    Instantiating a configuration with the defaults will yield a similar configuration to that of the Roberta
```
Comment (NielsRogge) [inline src/transformers/models/roberta/configuration_roberta.py:41]:
```suggestion
    Instantiating a configuration with the defaults will yield a similar configuration to that of the RoBERTa
```

File: src/transformers/models/squeezebert/configuration_squeezebert.py [hunk 13]
```diff
@@ -30,7 +30,9 @@
 class SqueezeBertConfig(PretrainedConfig):
     r"""
     This is the configuration class to store the configuration of a [`SqueezeBertModel`]. It is used to instantiate a
-    SqueezeBERT model according to the specified arguments, defining the model architecture.
+    SqueezeBERT model according to the specified arguments, defining the model architecture. Instantiating a
+    configuration with the defaults will yield a similar configuration to that of the SqueezeBert
```
Comment (NielsRogge) [inline src/transformers/models/squeezebert/configuration_squeezebert.py:34]:
```suggestion
    configuration with the defaults will yield a similar configuration to that of the SqueezeBERT
```

File: src/transformers/models/xlm_roberta/configuration_xlm_roberta.py [hunk 14]
```diff
@@ -37,7 +37,8 @@
 class XLMRobertaConfig(RobertaConfig):
     """
     This class overrides [`RobertaConfig`]. Please check the superclass for the appropriate documentation alongside
-    usage examples.
+    usage examples. Instantiating a configuration with the defaults will yield a similar configuration to that of the
+    XLMRoberta [xlm-roberta-base](https://huggingface.co/xlm-roberta-base) architecture.
```
Comment (NielsRogge) [inline src/transformers/models/xlm_roberta/configuration_xlm_roberta.py:41]:
```suggestion
    XLMRoBERTa [xlm-roberta-base](https://huggingface.co/xlm-roberta-base) architecture.
```

**Review by NielsRogge — APPROVED** [review]
Thanks a lot for this PR, awesome that this gets improved.

Left some comments, just for consistency, I would always use the template:

> "will yield a similar configuration of that of the - snake-cased model name -  [checkpoint name](link) architecture".

File: src/transformers/models/speech_to_text_2/configuration_speech_to_text_2.py (hunk 1 above)
Comment (patrickvonplaten) [reply src/transformers/models/speech_to_text_2/configuration_speech_to_text_2.py:34]: Great catch!

**Review by patrickvonplaten — APPROVED** [review]
Looked at all the speech models - looks good to me!

**Pull request merged**